src/Microsoft.ML.Transforms/TensorflowTransform.cs

```diff
@@ -0,0 +1,681 @@
+        }
+
+        private ITensorValueGetter CreateTensorValueGetterVec(IRow input, TFDataType tfType, ColumnType type, int colIndex, TFShape tfShape)
+        {
+            if(tfType == TFDataType.Float)
+            {
+                return CreateTensorValueGetter<float>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Double)
+            {
+                return CreateTensorValueGetter<double>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Int32)
+            {
+                return CreateTensorValueGetter<DvInt4>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Int64)
+            {
+                return CreateTensorValueGetter<Int64>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.Bool)
+            {
+                return CreateTensorValueGetter<bool>(input, type, colIndex, tfShape);
+            }
+
+            if (tfType == TFDataType.String)
+            {
+                return CreateTensorValueGetter<string>(input, type, colIndex, tfShape);
+            }
+
+            throw Host.ExceptNotSupp("Tensorflow type not supported");
+        }
+
+        private ITensorValueGetter[] GetTensorValueGetters(IRow input, int iinfo)
+        {
+            var info = _bindings.Infos[iinfo];
+            var tfInfo = _bindings.TfColInfo[iinfo];
+            var srcTensorGetters = new ITensorValueGetter[info.SrcIndices.Length];
+            for (int j = 0; j < info.SrcIndices.Length; j++)
+            {
+                int colIndex = _bindings.Infos[iinfo].SrcIndices[j];
+                srcTensorGetters[j] = CreateTensorValueGetterVec(input, tfInfo.TfTypes[j], info.SrcTypes[j], colIndex, tfInfo.TfShapes[j]);
+            }
+            return srcTensorGetters;
+        }
+
+        private unsafe T[] FetchData<T>(IntPtr data, int size)
+        {
+            var result = new T[size];
+
+            GCHandle handle = GCHandle.Alloc(result, GCHandleType.Pinned);
+            IntPtr target = handle.AddrOfPinnedObject();
+
+            Int64 sizeInBytes = size * Marshal.SizeOf((typeof(T)));
+            Buffer.MemoryCopy(data.ToPointer(), target.ToPointer(), sizeInBytes, sizeInBytes);
+            handle.Free();
+            return result;
+        }
+
+        private Delegate MakeGetter(IRow input, int iinfo)
+        {
+            var info = _bindings.Infos[iinfo];
+            var outInfo = _bindings.OutputCols[iinfo];
+            var tfType = _bindings.OutputTFTypes[iinfo];
+
+            if (tfType == TFDataType.Float)
+            {
+                Func<IRow, int, ValueGetter<VBuffer<float>>> srcDel = MakeGetter<float>;
+                var meth = srcDel.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(outInfo.ItemType.RawType);
+                return (Delegate)meth.Invoke(this, new object[] { input, iinfo });
```

Comment:
You can make this call only once for all types.
Also, there is a utility method Utils.MarshalInvoke that does this, and provides compile time type checking.